lib/msf/core/db_manager/import/metasploit_framework/zip.rb

```diff
@@ -236,4 +242,8 @@ def import_msf_zip(args={}, &block)
       import_msf_collateral(new_args)
     end
   end
+
+  def is_child_of?(target_dir, target)
+    target.match?(/^#{target_dir}/)
```

Comment:
`target.downcase.start_with?(target_dir.downcase)` might be a better option.

Reply:
Isn't this check bypassed if if the target is `target_dir/../file`?
You might need to call 'cleanpath' from the Pathname object to normalize the specified path:

```
2.6.1 :006 > Pathname.new("cow/../dog").to_s
 => "cow/../dog" 
2.6.1 :007 > Pathname.new("cow/../dog").cleanpath.to_s
 => "dog" 
```

Reply:
I was working under the assumption that the `target_dir` is an absolute path, because it's generated on [Line 169](https://github.com/rapid7/metasploit-framework/blob/73ad312dbfac7ef9c67fa6c6b4fb30e97dd9349e/lib/msf/core/db_manager/import/metasploit_framework/zip.rb#L169) .  In the case of absolute path, the current form of
`(Pathname.new("/cow/../dog") + '../../../../../goat').to_s`
has a `cleanpath` behavior.
```
dropping into irb from msfconsole:
>> target_dir = Dir::tmpdir
=> "/var/folders/tc/597c6qh15hvf2cvrgmdgcsw19hn4xh/T"
>> (Pathname.new('/cow/dog/cat') + '../../../../../goat').to_s
=> "/goat"
```

I can change it to
```
>> File.expand_path('cow/dog/cat' + '../../../../../goat', '/').to_s
=> "/goat"
>> File.expand_path('/cow/dog/cat' + '../../../../../goat', '/').to_s
=> "/goat"
```
to cover the case of `target_dir` being specified as a relative path.  And it also makes the code more explicit and better shows the intent, which I think is helpful for others reading the code

Reply:
Interesting, the behavior doesn't happen in plain IRB outside of Metasploit. Does Metasploit actually somehow monkeypatch/change the stock behavior for Pathname? Or is the fact that I'm testing this on Ruby 2.4.2 indicate that this is a Ruby-version dependent behavior, and we should explicitly call cleanpath anyway to make it consistent?

```
2.4.2 :004 > (Pathname.new("/cow/../dog") + '../../../../../goat').to_s
 => "/cow/../../../../../goat" 
```

Reply:
I also observed that `Dir::tmpdir` doesn't work in regular IRB.  I suspect it _is_ monkey patched somewhere in Metasploit

Reply:
Looks like we're using `tmpdir` from the stdlib:
```
>> Dir.method(:tmpdir).source_location
=> ["/Users/sgonzalez/.rvm/rubies/ruby-2.6.2/lib/ruby/2.6.0/tmpdir.rb", 21]
```

Reply:
from regular IRB:
```
ruby-2.6.2@metasploit-framework :001 > require 'tmpdir'
true
ruby-2.6.2@metasploit-framework :002 > Dir::tmpdir
"/var/folders/tc/597c6qh15hvf2cvrgmdgcsw19hn4xh/T"
```